### ISSUE TITLE
[Feature request] Easily differentiate between chat contexts or users (close friends vs professional for example)

### ISSUE THREAD

**Issue by urbanhusky** [telegramdesktop/tdesktop#2642]
I think this could partially be one of the actual feature requests behind the "background per contact" requests.

As a user, I'd like to quickly be able to differentiate between the group of user or single users, so that I don't end up mixing professional contacts, close friends, family, partner, acquaintances etc.
Example of such issues range from sending `whatcha doing tonight? <3` to a professional or business contact, or more trivial things such as sending links to the family album to your board gaming group (further examples can be extrapolated with a little bit of creativity - especially if you assume that your intended target was your partner).

Currently, all I see is the chat history, which looks the same for every chat (with only easily overlooked details such as profile pictures) and the selected chat on the left hand side. My focus is on the chat itself and the shape of the chat - and even more on where I can type messages, send photos or possibly very inconvenient stickers.

One such way to help prevent accidentally sending messages, images etc. to the wrong contact could be to provide options to set background images or colours, or chat box styles per user (or even for multiple users at the same time).

**Comment by Innomen** [comment]
Agreed completely. This problem essentially constitutes a security risk.

**Comment by stek29** [comment]
Would it really help? I mean, If you can't understand that it's wrong chat by title, profile picture and message history, would different backgrounds or bubbles make any difference?

**Comment by Innomen** [comment]
@stek29 
Putting aside the gross and unwarranted insult...

To answer your question simply. Yes it would make a substantial difference. And the attempted shaming of people who disagree is a lousy (or nonexistent) counter argument.

**Comment by urbanhusky** [comment]
Imagine having a large enough display - say 22", 26" or even more. I run Telegram in a very big window so the list of chats on the left is ~20cm from where I'm looking at, and the title bar is ~24cm from where I'm typing. This blurs out any information that might be present there because it's only peripheral vision - and the shape, text etc. are things which most people could not even be aware of I'd wager.

Similar to how I don't see the title bar or tabs in a browser window - the only way to get my attention there would be to animate or change things, which is why applications often request focus, blink or use badges to grab your attention - and why ads are most often animated or otherwise don't fit the context.

The message history is mostly just an arrangement of chat bubbles. Most chats involve both parties, so the shape of the chat is usually not calling attention to the history or context (unless any of you really spam the heck out of specific stickers). Let's be honest - I'm not going to read the past history when I'm actively juggling multiple conversations.

**Comment by stek29** [comment]
@Innomen I didn't mean to offend anyone. I may be rude, but that's me, sorry.
Also, I'm also having this issue -- I may send something to completely wrong person.
But what I highly doubt that custom backgrounds would help.

IMO It would be better to have some groups, and assign them some icons.
And that icon may be displayed next to message input box.

But I never used TDesktop on big displays, and after @urbanhusky's comment I may understand why backgrounds would be useful too.

**Comment by urbanhusky** [comment]
Groups would be a nice thing to have, if switching between them is not too difficult or too much effort. 
Backgrounds per chat (or group) would both be a nice customisation feature and might help mitigate the issue of mixed up conversations in a very apparent way. 

I'm by no means settled on an exact way how this could be solved and I'm glad that the problem is now better understood :)

**Comment by Innomen** [comment]
@stek29 

All's well so long as you see our point now. (Even if you disagree with it.)

**Comment by dmrsol** [comment]
Good idea! I would not be limited to a simple opportunity to change the background. it is best to allow the dialogue to assign any group. The group could be applied notification settings on all chats, for example.

**Comment by urbanhusky** [comment]
*Group* is a bit of an ambivalent term in this discussion. Telegram has groups (as in group chat).
The idea of being able to categorise users and groups (chats) do differentiate between them should not really be called a *group*.

**Comment by AndydeCleyre** [comment]
I think @urbanhusky is suggesting we transition to a less ambiguous term such as "class" or "flavor" or "context" (I expect someone will come up with something better than those).

**Comment by stek29** [comment]
@AndydeCleyre yup, I just wanted to clarify for people who may misunderstand what was meant by group in this discussion.
Context is the best, but doesn't show the point really.

**Comment by urbanhusky** [comment]
I want to avoid the term *group* because it is already heavily used in telegram (just search for *telegram group*) and might lend itself to hasty and wrong interpretations.

These two concepts are at the core of this feature request:
* We already have conversations or chats - which can be either the correspondence with a single user or a whole group
* We desire to be able to easily distinguish between them (not just chat vs. group but between each single chat/group), which we could also require to be grouped together via user-defined contexts (such as: family, professional, board game night, ...). Or to put it more mathematically: we want to be able to partition the chats we are having and quickly and easily be able to identify which partition the current chat window is part of.

Please don't treat the words I've used here as gospel :)

**Comment by Innomen** [comment]
This really is the essence of the different backgrounds request I made. If I could merge the two requests I would. This one is much more clear and deserves whatever support mine received.

Instant and obvious differentiation is the goal, as stated in the first line of this thread.

**Comment by devcline** [comment]
+1

**Comment by navossoc** [comment]
> Backgrounds per chat (or group) would both be a nice customisation feature and might help mitigate the issue of mixed up conversations in a very apparent way.

I did it today, shame on me.

The same person sent me a message before on private message, then in a group.
I mixed up the chats and didn't notice...

+1 for this request.
at least a custom background for group chats.

**Comment by damir98** [comment]
+1 for this request.
And completly agree for custom background per chat as a solve direction.

**Comment by patoshii** [comment]
same here. accidentally posted in a group chat. 

custom backgrounds for each group and contact!!!!!!

**Comment by javiereduardo** [comment]
since 2015 is a request... but nothing happens in telegram :(

**Comment by patoshii** [comment]
Has the telegram dev's actually put this on their radar to implement?

**Comment by devcline** [comment]
maybe or maybe not, its the same story with [5 limit pins](https://github.com/telegramdesktop/tdesktop/issues/4037) ...

**Comment by kenorb** [comment]
Related: #1107, #1505, #2702, #4179, #4873, #5560

**Comment by devcline** [comment]
I have a temps solution that i use. **Have a second number and use TELEGRAM for Contacts and TELEGRAM X for Groups and Bots.**

**Comment by urbanhusky** [comment]
> I have a temps solution that i use. **Have a second number and use TELEGRAM for Contacts and TELEGRAM X for Groups and Bots.**

Aside from your solution being _don't use Telegram_, it doesn't solve the boss vs girlfriend issue :)

**Comment by AndydeCleyre** [comment]
Additionally, this is a reported issue guy the desktop client, and AFAIK there's no Telegram X desktop client. As an aside,  I would like to know if X will be open sourced, or even has a public issue tracker.

**Comment by D0ctorWh0** [comment]
We definitely need that feature

**Comment by kenorb** [comment]
Vote for it at https://t.me/tgbeta/3500

**Comment by RememberTheAir** [comment]
A bit disappointed that this repo's admins linked the official Telegram poll on a third-party channel instead of mentioning the original source, which is [this message in @telegram](https://t.me/telegram/105). Not that it changes anything, but still

**Comment by john-preston** [comment]
@RememberTheAir What admins?

**Comment by RememberTheAir** [comment]
@john-preston mmm sorry, "admins" is not the correct word. I was talking about [this comment](https://github.com/telegramdesktop/tdesktop/issues/1107#issuecomment-475874410) in a similar issue where the poll has already been mentioned by Auchri. Just ignore my comment, dpesn't really contribute to the discuccion

**Comment by doodoo787** [comment]
THIS!!!

I've been searching all over the internet for a solution to this problem. I just want to easily differentiate between groups -- which get's hard especially if you have certain people that take part in several of your groups.

PROBLEM:
e.g. if your spouse is in your private two-person-chat, in your kids socker group, in your happy-go-lucky-lets-go-party-friends-group AND your official-work-group it get's pretty ugly pretty quick.

Cause if she sends some ambigious text and your head decides it belongs to the wrong conversation ... well I guess you can imagine all the ways that could go wrong.

PROBLEM2:
This is even more of an issue in the iOS app!!
Because the group icon is way at the top and I don't even have the left column indicating which chat I'm using (which doesn't really help on the big screen of the mac, but I don't even have that).

NOT A SOLUTION:
Oh and icons next to the text-input-window alone won't do the trick. Since I am not in controll of those pictures. I already have problems distinguishing tinsy tiny profile pics let alone group pictures that look mostly the same anyways.

SOLUTION
**Custom chat-backgrounds (which I myself can choose!) seems to be the easiest solution to this MAJOR security problem!!!**

However, I will gladly take ANY kind of solution to this!!!

**Comment by doodoo787** [comment]
Oh my gosh this has been pending for THREE YEARS??

**Comment by Ansem93** [comment]
There is any hope to see this thing implemented? I have too many groups, bots and channels. I really need tabs :(

**Comment by kenorb** [comment]
Using a desktop version, 5.7 (181196) Stable, clicking on the search box, it separates people and recent messages quite nicely. But it's not possible in v1.8.

I think this feature should be added into v1.8.

**Comment by RememberTheAir** [comment]
"5.7" is the [native MacOS app](https://macos.telegram.org/), not telegram desktop. Just for context

**Comment by Ansem93** [comment]
I think that the solution of "Telegreat" (or "plus messenger" for android) is the best. Simple tabs that divide users, groups, channels and bots.

**Comment by joesixpack** [comment]
You mean like this?  Use Bettergram.  TG obviously doesn't see it as a priority after four years.

![image](https://user-images.githubusercontent.com/23145642/67513038-e61c3100-f64e-11e9-9c12-9f3708ba1630.png)

**Comment by kenorb** [comment]
Chat folders has been implemented now. See: https://telegram.org/blog/folders

> Telegram was updated to version 6.0 (196005)
>
> Chat Folders and More
> 
> • Organize chats into Chat Folders if you have too many chats.
> 
> • Create custom folders with flexible settings, or use default recommendations.
> 
> • Pin an unlimited number of chats in each folder.
> 
> • Archive chats to hide them from "All chats". Muted chats will always stay in the Archive.

**Comment by yashpalgoyal1304** [comment]
> Chat folders has been implemented now

So, the thing, as this issue stated, left is customisation of chat bg per folder right?

**Comment by kenorb** [comment]
> > Chat folders has been implemented now
> 
> So, the thing, as this issue stated, left is customisation of chat bg per folder right?

Yep, something like:

![image](https://user-images.githubusercontent.com/266306/88527614-6b28e200-cff5-11ea-9b91-9a9ce7336e13.png)

**Comment by urbanhusky** [comment]
> 
> 
> Hey there!
> 
> This issue will be automatically closed in 7 days if there would be no activity. We therefore assume that the user has lost interest or resolved the problem on their own.
> 
> Don't worry though; if this is an error, let us know with a comment and we'll be happy to reopen the issue.
> 
> Thanks!

This is still, very much, relevant.

**Comment by doodoo787** [comment]
The new grouping feature is nice.
However, it doesn't solve the problem at heart:

I open my messages, I am **not aware** which chat I'm in (my brain just doesn't keep very good track of whom I texted with last).

**The only thing** that would allow me to quickly know if I'm texting with my boss or my fiancé would be if there was a different background for each of them. (Make it optional, not everybody has a scatterbrain like me).

And yes, I can look at the icon. But if **I think I know** who I'm texting with, I won't do it and end up asking my boss for the color of his panties (never happened -- honestly).
However, I would realize a pink background against a black one immediately (and before sending anything embarrassing).

Please don't close this topic and give us customizable backgrounds **per chat**.
Thank you very much.

**Comment by Xtremis77** [comment]
Perhaps with the new users influx, there might be some investment in this feature.

**Comment by dienluong** [comment]
Four years and this still can't be done, while countless more involving features have been added.  Very puzzling.

**Comment by Aokromes** [comment]
4.8 ships 
- Set a custom wallpaper for any 1-on-1 chat.
idk if that can help you to close this ticket.

**Comment by Vargock** [comment]
I think this is still relevant, especially for Telegram Desktop.

Folders help, but when Telegram is used for work, even one folder can contain support chats, dev chats, bot alerts, urgent channels, and regular discussions. In the chat list they still look very similar, so it takes extra attention to find the right one.

A simple optional visual marker would help a lot: per-chat background row color, a colored left border, or stronger folder-based color markers.

There is also a related suggestion on Telegram Bugs:
https://bugs.telegram.org/c/50197
